src/expr/src/scalar/func.rs

```diff
@@ -7907,6 +7950,12 @@ impl VariadicFunc {
     pub fn could_error(&self) -> bool {
         match self {
             VariadicFunc::And | VariadicFunc::Or => false,
+            VariadicFunc::Coalesce => false,
+            VariadicFunc::Concat | VariadicFunc::ConcatWs => false,
+            VariadicFunc::Replace => false,
+            VariadicFunc::Translate => false,
+            VariadicFunc::ArrayIndex { .. } => false,
+            VariadicFunc::ListCreate { .. } | VariadicFunc::RecordCreate { .. } => false,
```

Comment:
Maybe you could add `Greatest` and `Least`. Those sometimes occur in complex timestamp manipulations.

Reply:
Sure... it's not obvious from the types that those functions never throw, but looking at the implementations it does seem like the only way they might return an error is if they get one as an input.